frontend/src/renderer/pages/config/Config.tsx

```diff
@@ -89,18 +88,39 @@ function Config({ setConfigValid }) {
         },
     });
 
-    // Add useEffect to initialize form values from localStorage
     useEffect(() => {
+        // Make an API call to fetch the configuration data only on the first load
+        if (!dataLoaded) {
+            apiCall('/config', 'GET', null, true)
+                .then((response) => {
+                    const fetchedConfig = response.data;
+                    // Set the form values based on the retrieved data
+                    form.setValues({
+                        OPENAI_API_KEY: fetchedConfig.openai_api_key || '',
+                        model: fetchedConfig.model || '',
+                        embeddingsmodel: fetchedConfig.embeddings_model || '',
+                    });
+                    // Save to localStorage for future loads
+                    localStorage.setItem('config', JSON.stringify(fetchedConfig));
+                    // Set the dataLoaded flag to true to prevent future API calls
+                    setDataLoaded(true);
+                })
+                .catch((error) => {
+                    console.log('Error fetching config:', error);
+                    // Handle the error as needed, maybe show a notification or take other actions
+                });
+        }
+
```

Comment:
this would make the `first load` everytime one goes back to Config component - is the if `dataLoaded` then even needed? as the state will be gone if we change the component.